app/assets/javascripts/controllers/automation_manager/automation_manager_form_controller.js

```diff
@@ -65,14 +64,27 @@ ManageIQ.angular.app.controller('automationManagerFormController', ['$http', '$s
     miqService.sparkleOff();
   }
 
+  vm.validateClicked = function($event, authType, url, formSubmit) {
+    console.log('data: ', $event, authType, url, formSubmit);
+    miqService.validateWithREST($event, authType, url, formSubmit)
+      .then(function success(data) {
+        console.log('data');
+        if (data.level === 'error') {
+          $scope.angularForm.authCredentialsForm.default_auth_status.$setViewValue(false);
+        } else {
+          $scope.angularForm.default_auth_status.$setViewValue(true);
+        }
+        miqService.miqFlash(data.level, data.message);
+        miqService.sparkleOff();
+      });
+  };
+
```

Comment:
Why was this function moved from `configurationManagerService` to here?

Reply:
The main reason why i decided to move this function bac to controller is, that the validation method requires `angularForm` and directly is mutating `viewValue`. I don't think that any function in any service should ever do something like that. Validation is done via the `validateWithRest` and any change to controller view or model should be handled by controller itself. Also i find the `$setViewValue(...)` a bit *hacky* solution and it is not supported in angular 2+. Now it is chnaged directly in form model. It is also expecting, that there will be some `default_auth_status`, which is true for now, but as soon as someone will use some different form structure or even element name, this will break. That is why i moved this back to controller.